core/src/com/unciv/ui/components/fonts/DiacriticSupport.kt

```diff
@@ -0,0 +1,196 @@
+ *  - We'll deal with one glyph at a time arranges left to right, and expect a finite number of combination glyphs (all fit into the Unicode Private Use Area **together** with [FontRulesetIcons]).
+ *  - We'll recognize these combos in the translated texts at translation loading time and map each combo into a fake alphabet, which fulfills the "one Char one Glyph" tenet.
+ *  - Conversely, the loader will build a map of distinct combinations -codepoint sequences- that map into a single glyph and correlate each with their fake alphabet codepoint.
+ *  - At render time, only the map of fake alphabet codepoints to their original codepoint sequences is needed.
+ *        * Remember, NativeBitmapFontData renders and caches glyphs on demand
+ *        * GlyphLayout (a Gdx class) needs a Glyph that's not yet cached, then:
+ *        * If it's in the fake alphabet, we'll ask the OS to render the original codepoint sequence instead
+ *        * Otherwise render the single Char as before
+ *
+ *  ### Usage
+ *  - Call [reset] when translation loading starts over
+ *  - Call [prepareTranslationData] once a translation file is read (their map of key (left of =) to translation (right of =) is in memory, pass that as argument)
+ *  - Check [noDiacritics] - if true, the rest of that language load need not bother with diacritics
+ *  - Call [remapDiacritics] on each translation and store the result instead of the original value
+ *  - If you wish to save some memory, call [freeTranslationData] after all required languages are done
+ *  - Later, [NativeBitmapFontData.createAndCacheGlyph] will use [getStringFor] to map the fake alphabet back to codepoint sequences
+ *
+ *  ### Notes
+ *  - [FontRulesetIcons] initialize ***after*** Translation loading. If this ever changes, this might need some tweaking.
+ */
+object DiacriticSupport {
+    /** Start at end of Unicode Private Use Area and go down from there: UShort is the preferred Char() constructor parameter! */
+    private const val startingReplacementCodepoint: UShort = 63743u // 0xF8FF
+
+    private object TranslationKeys {
+        const val range = "diacritics_joinable_range"
+        const val left = "left_joining_diacritics"
+        const val right = "right_joining_diacritics"
+        const val joiner = "left_and_right_joiners"
+    }
+
+    //region The following fields can persist over loading multiple languages
+    private var nextFreeDiacriticReplacementCodepoint = startingReplacementCodepoint
+    private val fakeAlphabet = mutableMapOf<Char, String>()
+    private val inverseMap = mutableMapOf<String, Char>()
+    //endregion
+
+    //region These fields will only persist during one language load
+    private var noDiacritics = true
+    private val charClassMap = mutableMapOf<Char, CharClass>()
+    private var defaultCharClass = CharClass.None
+
+    private class LineData(capacity: Int) {
+        val output = StringBuilder(capacity)
+        val accumulator = StringBuilder(8)
```

Comment:
`StringBuilder(8)`, does this means the limit is 8? will not work. I guess you based this on https://en.wikipedia.org/wiki/Diacritic#Orthographic. But if you notice, below there is a list of 9 characters and 8 diacritics. So, the sequence we want is already 9 characters long.

So, this will not work as the number of characters in combinations including normal characters can be much higher.